server/market/bookrouter.go

```diff
@@ -225,13 +223,13 @@ out:
 					epochNote.OrderType = msgjson.CancelOrderNum
 					epochNote.TargetID = o.TargetOrderID[:]
 				}
+
 				epochNote.Seq = seq
 				epochNote.MarketID = book.name
-				t := uint64(u.order.Time())
-				epochNote.Epoch = t - t%r.epochLen
+				epochNote.Epoch = uint64(u.epochIdx)
```

Comment:
The source of the order feed knows this info, so now we just send it so the BookRouter does not have to know epoch durations or inspect order time stamps.